src/ArchiveExtractor/include/ArchiveExtractor/ArchiveExtractor.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "libzippp.h"
+#include <memory>
+#include <string>
+#include <boost/filesystem.hpp>
+
+using namespace libzippp;
+
+class ArchiveExtractor
+{
+  public:
+    virtual std::vector<ZipEntry> readEntries(const std::string &source) = 0;
+    virtual void writeEntry(const std::string &outputFilename, const std::string &textData) = 0;
```

Comment:
`readEntries` and `writeEntry` both seem more like an implementation detail not part of interface. Cant they be private members of `EpubExtractor`?

Reply:
I was thinking that any kind of archive extractor that will inherit from the interface base class should support the `readEntries` and `writeEntry` functionality. That's why I made them part of the interface. What do you think?

Reply:
Good point 👍